File: code-review/tests/fixtures/vulnerable_sample.js
// Deliberate triggers for JS-001..JS-006. Parsed only, never executed.
const { exec } = require("child_process");
const https = require("https");

function runExpr(expr) {
  return eval(expr);                                // JS-001
}

function schedule() {
  setTimeout("runIt()", 100);                        // JS-001
}

function makeRunner(src) {
  return new Function(src);                          // JS-001
}

function render(el, data) {
  el.innerHTML = data;                                // JS-002
}

function writePage(name) {
  document.write("<b>" + name + "</b>");              // JS-002
}

const apiKey = "sk-live-not-a-real-key-000000000000"; // JS-003
const config = { password: "hunter2xyz" };            // JS-003

function backup(path) {
  exec("tar czf backup.tgz " + path);                 // JS-004
}

function makeToken() {
  const sessionToken = Math.random().toString(36).slice(2); // JS-005
  return sessionToken;
}

const agent = new https.Agent({ rejectUnauthorized: false }); // JS-006
process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";                // JS-006

File: code-review/tests/fixtures/vulnerable_sample.c
/* Deliberate triggers for C-001..C-006. Parsed only, never compiled/run. */
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define API_KEY "abc123xyz999_not_real"           /* C-003 */

char *g_password = "hunter2xyz";                  /* C-003 */

void read_input(char *buf) {
    gets(buf);                                     /* C-001 */
}

void copy_name(char *dst, const char *src) {
    strcpy(dst, src);                              /* C-001 */
}

void log_message(const char *user_input) {
    printf(user_input);                            /* C-002 */
}

void run_backup(const char *path) {
    char cmd[256];
    sprintf(cmd, "tar czf backup.tgz %s", path);    /* C-001 */
    system(cmd);                                    /* C-004 */
}

int make_token(void) {
    int session_token = rand();                     /* C-005 */
    return session_token;
}

char *scratch(int n) {
    return alloca(n);                                /* C-006 */
}
